### PR TITLE
removed Thread.sleep() and added SiddhiTesthelper

### PR DESCRIPTION
**Pull request by AnuGayan**

### CODE REVIEW

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java [hunk 1]
```diff
@@ -61,15 +73,37 @@ public void testGeometry() throws Exception {
...
+                    if (count.get() == 2) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 3) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+
```
Comment (NisalaNiroshana) [inline component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java:91]: Please remove unnecessary new line

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java [hunk 2]
```diff
@@ -61,15 +73,37 @@ public void testGeometry() throws Exception {
...
+                    }
+                    if (count.get() == 4) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+
```
Comment (NisalaNiroshana) [inline component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java:97]: Please remove unnecessary new line

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java [hunk 3]
```diff
@@ -61,15 +73,37 @@ public void testGeometry() throws Exception {
...
+                    }
                 }
             }
         });
         siddhiAppRuntime.start();
         generateEvents(siddhiAppRuntime);
-        Thread.sleep(1000);
-        AssertJUnit.assertEquals(expectedResult.size(), eventCount);
+        SiddhiTestHelper.waitForEvents(100, 4, count, 60000);
```
Comment (NisalaNiroshana) [inline component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java:104]: I think we need to stop the siddhiAppRuntime. WDYT?

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java [hunk 4]
```diff
@@ -70,15 +79,46 @@ public void testPoint() throws Exception {
...
                 for (Event event : inEvents) {
                     logger.info(event);
-                    Boolean isWithin = (Boolean) event.getData(0);
-                    AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                    count.incrementAndGet();
+                    if (count.get() == 1) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+
```
Comment (NisalaNiroshana) [inline component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java:87]: Please remove the unnecessary new line

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java [hunk 5]
```diff
@@ -70,15 +79,46 @@ public void testPoint() throws Exception {
...
+                    }
                 }
             }
         });
         siddhiAppRuntime.start();
         generateEvents(siddhiAppRuntime);
-        Thread.sleep(1000);
-        AssertJUnit.assertEquals(expectedResult.size(), eventCount);
+        SiddhiTestHelper.waitForEvents(100, 6, count, 60000);
+        AssertJUnit.assertEquals(6, count.get());
```
Comment (NisalaNiroshana) [inline component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java:120]: Don't we need to stop the siddhiAppRuntime ? Please check the other test cases also

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java (hunk 1 above)
Comment (AnuGayan) [reply component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java:91]: Fixed by 26e6366

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java (hunk 2 above)
Comment (AnuGayan) [reply component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java:97]: Fixed by 26e6366

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java (hunk 3 above)
Comment (AnuGayan) [reply component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java:104]: Fixed by 26e6366

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java (hunk 4 above)
Comment (AnuGayan) [reply component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java:87]: Fixed by 26e6366

File: component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java (hunk 5 above)
Comment (AnuGayan) [reply component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinTestCase.java:120]: Fixed by 26e6366

**Pull request merged**